.github/workflows/release-llvm.yml

```diff
@@ -0,0 +1,205 @@
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rust-src
+          target: wasm32-unknown-emscripten
+
+      - name: versions
+        run: |
+          rustup show
+          cargo --version
+          cmake --version
+          echo "bash:" && bash --version
+          echo "ninja:" && ninja --version
+          echo "clang:" && clang --version
+
+      - name: Build host LLVM
+        run: |
+          make install-llvm
+
+      - name: Build musl LLVM
+        run: |
+          revive-llvm --target-env musl clone
+          revive-llvm --target-env musl build --llvm-projects lld --llvm-projects clang
+
+      - name: Build emscripten LLVM
+        run: |
+          revive-llvm --target-env emscripten clone
+          source emsdk/emsdk_env.sh
+          revive-llvm --target-env emscripten build --llvm-projects lld
+
+      - name: clean
+        # check removed files
+        run: |
+          for target in gnu emscripten musl; do 
+            cd target-llvm/${target}/target-final/bin/
+            rm diagtool llvm-libtool-darwin llvm-lipo llvm-pdbutil llvm-dwarfdump llvm-nm llvm-readobj llvm-cfi-verify \
+               sancov llvm-debuginfo-analyzer llvm-objdump llvm-profgen llvm-extract llvm-jitlink llvm-c-test llvm-gsymutil llvm-dwp \
+               dsymutil llvm-dwarfutil llvm-exegesis lli clang-rename bugpoint clang-extdef-mapping clang-refactor c-index-test \
+               llvm-reduce llvm-lto clang-linker-wrapper llc llvm-lto2
+            cd -
+          done
+
+      - name: package artifacts
+        run: |
+          tar -cf "${{ github.ref_name }}-linux-gnu.tar.gz" target-llvm/gnu/target-final
+          tar -cf "${{ github.ref_name }}-emscripten.tar.gz" target-llvm/emscripten/target-final
+          tar -cf "${{ github.ref_name }}-musl.tar.gz" target-llvm/musl/target-final
+
+      - name: upload archive to release
+        uses: softprops/action-gh-release@v2
+        with:
+          files: |
+            ${{ github.ref_name }}-linux-gnu.tar.gz
+            ${{ github.ref_name }}-emscripten.tar.gz
+            ${{ github.ref_name }}-musl.tar.gz
+
+#  build-windows-gnu:
+#    needs: create-release
+#    runs-on: windows-latest
+#    env:
+#      MSYSTEM: "MINGW"
+#      RUST_LOG: trace
+#    steps:
+#      - uses: actions/checkout@v4
+#
+#      - name: install win deps
+#        run: |
+#          choco install ninja
+#
+#      - name: Setup msys2
+#        id: msys2
+#        uses: msys2/setup-msys2@v2
+#        with:
+#          path-type: inherit
+#          install: >-
+#            base-devel
+#            mingw-w64-x86_64-clang
+#            mingw-w64-x86_64-lld
+#            mingw-w64-x86_64-rust
+#            mingw-w64-x86_64-cmake
+#            mingw-w64-x86_64-gcc
+#            mingw-w64-x86_64-gcc-libs
+#            mingw-w64-x86_64-python
+#            mingw-w64-clang-x86_64-riscv64-unknown-elf-toolchain
```

Comment:
Use msvc since you are building on windows. It is the most compatible and easiest to build for natively. No reason to pull mingw.

Reply:
As a matter of fact, I've started with msvc and failed. The suggestion from @xermicus was to look for inspiration in how matter-labs do their windows releases, and [they use mingw](https://github.com/matter-labs/era-compiler-ci/blob/main/.github/actions/prepare-msys/action.yml)

Reply:
I think [this](https://github.com/llvm/llvm-project/issues/47189) was the issue that I've been hitting with MSVC

Reply:
> I would be really surprised if nobody could build LLVM with the Microsoft toolchain.

I wouldn't be surprised if nobody has built matterlabs fork of LLVM with MS toolchain, as they're using mingw themselves.  
Given the sort of issue I'm getting right now, however, I'm inclined to give it another go.